wagtail/admin/rich_text/converters/html_to_contentstate.py

```diff
@@ -0,0 +1,288 @@
+from html.parser import HTMLParser
+import re
+
+from wagtail.admin.rich_text.converters.contentstate_models import (
+    Block, ContentState, Entity, EntityRange, InlineStyleRange
+)
+from wagtail.admin.rich_text.converters.html_ruleset import HTMLRuleset
+from wagtail.core.rich_text import features as feature_registry
+from wagtail.core.models import Page
+
+
+class HandlerState(object):
+    def __init__(self):
+        self.current_block = None
+        self.current_inline_styles = []
+        self.current_entity_ranges = []
+        # what to do with leading whitespace on the next text node we encounter: strip, keep or force
+        self.leading_whitespace = 'strip'
+        self.list_depth = 0
+        self.list_item_type = None
+        self.pushed_states = []
+
+    def push(self):
+        self.pushed_states.append({
+            'current_block': self.current_block,
+            'current_inline_styles': self.current_inline_styles,
+            'current_entity_ranges': self.current_entity_ranges,
+            'leading_whitespace': self.leading_whitespace,
+            'list_depth': self.list_depth,
+            'list_item_type': self.list_item_type
+        })
+
+    def pop(self):
+        last_state = self.pushed_states.pop()
+        self.current_block = last_state['current_block']
+        self.current_inline_styles = last_state['current_inline_styles']
+        self.current_entity_ranges = last_state['current_entity_ranges']
+        self.leading_whitespace = last_state['leading_whitespace']
+        self.list_depth = last_state['list_depth']
+        self.list_item_type = last_state['list_item_type']
+
+
+class ListElementHandler(object):
+    """ Handler for <ul> / <ol> tags """
+    def __init__(self, list_item_type):
+        self.list_item_type = list_item_type
+
+    def handle_starttag(self, name, attrs, state, contentstate):
+        state.push()
+
+        if state.list_item_type is None:
+            # this is not nested in another list => depth remains unchanged
+            pass
+        else:
```

Comment:
I think it's important to keep the explanatory comment against the 'do nothing' case - otherwise, it's not really clear what the significance of `state.list_item_type` being `None` is.